File: GuestFlow.Frontend/src/types/supplier.ts
export interface Supplier {
  id: number;
  name: string;
  type: string;
  contactName?: string;
  phoneNumber?: string;
  email?: string;
  address?: string;
  website?: string;
  notes?: string;
  isActive: boolean;
  defaultCurrency?: string;
  defaultCost?: number;
  isDeleted: boolean;
  createdDate: string;
  updatedDate: string;
  createdBy?: string;
  updatedBy?: string;
}

export interface SupplierCost {
  id: number;
  supplierId: number;
  transferId?: number;
  cityTourId?: number;
  yachtTourId?: number;
  restaurantReservationId?: number;
  costAmount: number;
  currency: string;
  costType?: string;
  description?: string;
  validFrom?: string;
  validTo?: string;
  isActive: boolean;
  isDeleted: boolean;
  createdDate: string;
  updatedDate: string;
  createdBy?: string;
  updatedBy?: string;
}

export type SupplierType = 'Yacht' | 'Transfer' | 'Restaurant' | 'Activity' | 'Hotel' | 'General';

export type CostType = 'BaseCost' | 'AdditionalFee' | 'Commission' | 'FuelCost' | 'DriverFee' | 'EquipmentCost' | 'SeasonalAdjustment' | 'Discount';

File: GuestFlow.Domain/Entities/Core/Supplier.cs
using GuestFlow.Domain.Entities.Core;

namespace GuestFlow.Domain.Entities.Core
{
    public class Supplier : BaseEntity
    {
        public string Name { get; set; }
        public string Type { get; set; } // Yacht, Transfer, Activity, Restaurant, General
        public string? ContactName { get; set; }
        public string? PhoneNumber { get; set; }
        public string? Email { get; set; }
        public string? Address { get; set; }
        public string? Website { get; set; }
        public string? Notes { get; set; }
        public bool IsActive { get; set; } = true;

        // Default pricing
        public string? DefaultCurrency { get; set; } = "USD";
        public decimal? DefaultCost { get; set; }

        // Navigation properties - will be added when needed
        // public virtual ICollection<TransferSupplierCost> TransferCosts { get; set; }
        // public virtual ICollection<TourSupplierCost> TourCosts { get; set; }
    }

    public enum SupplierType
    {
        Yacht,
        Transfer,
        Activity,
        Restaurant,
        Hotel,
        General
    }
}